scripts/pre-deploy-guard.sh

```
@@ -0,0 +1,129 @@
+# Paths come from env so the script ships OSS-safe (no private paths committed).
+# dev build env default mirrors the deploy script (EVALUCHAT_DEPLOY_ENV_FILE).
+dev_env="${EVALUCHAT_DEPLOY_ENV_FILE:-$HOME/open-canvas/teaching-prototype/.env}"
+prod_env="${EVALUCHAT_PROD_ENV_FILE:-}"
+if [[ -r "$dev_env" && -n "$prod_env" && -r "$prod_env" ]]; then
+  diff_flags="$(comm -3 \
+    <(grep -E '^NEXT_PUBLIC_[A-Z_0-9]+=' "$dev_env" 2>/dev/null | sort -u) \
+    <(grep -E '^NEXT_PUBLIC_[A-Z_0-9]+=' "$prod_env" 2>/dev/null | sort -u) )"
+  if [[ -z "$diff_flags" ]]; then
+    ok "B3 flag parity" "dev and prod env NEXT_PUBLIC_* sets are identical"
+  else
+    warn "B3 flag parity" "NEXT_PUBLIC_* differs between dev and prod env files — a rebuild flips whichever the file holds:\n$(echo "$diff_flags" | sed 's/^/         /')"
```

Shellcheck (0.11.0)

[style] See if you can use ${variable//search/replace} instead.

(SC2001)